File: client/src/service/FavoriteProductService.js
import axios from "axios";
import {FAVORITE_PRODUCTS_API} from "./constants";

export const addFavoriteProduct = async (productId) => {
    return await axios.post(`${FAVORITE_PRODUCTS_API}?product-id=${productId}`)
        .then(response => response.data)
}

export const removeFavoriteProduct = async (productId) => {
    return await axios.delete(`${FAVORITE_PRODUCTS_API}?product-id=${productId}`)
        .then(response => response.data)
}

export const favoriteProducts = async () => {
    return await axios.get(FAVORITE_PRODUCTS_API)
        .then(response => response.data)
}

File: server/src/main/java/com/web/server/services/FavoriteProductService.java
package com.web.server.services;

import com.web.server.dto.response.ProductResponse;

import java.util.List;

public interface FavoriteProductService {
    void addFavoriteProduct(Integer productId, String userIpAddress);
    void removeFavoriteProduct(Integer productId, String userIpAddress);
    List<ProductResponse> favoriteProducts(String userIpAddress);
}
